index.js

```diff
@@ -1,97 +1,226 @@
     if (err) throw err;
-    var conf = JSON.parse(data);
-    var urls = conf.urls;
+    const conf = JSON.parse(data);
+    const urls = conf.urls;
 
-    for (var name in urls) {
-        var url = urls[name].url;
-        var interval = urls[name].interval;
-        createloop(name, url, interval);
+    for (const [name, params] of Object.entries(urls)) {
+        createLoop(name, params);
     }
 });
 
 /* Create a loop that's executed every x milliseconds */
-function createloop(name, url, interval) {
-    var requestLoop = setInterval(function () {
-        request({
-            url: url,
-            method: "GET",
+function createLoop(name, params) {
+    setInterval(function () {
+        const requestDate = new Date();
+        const dateBounds = {
+            minTimeOfDay: params.minTimeOfDay,
+            maxTimeOfDay: params.maxTimeOfDay,
+            daysOfTheWeek: params.daysOfTheWeek
+        }
+
+        if (!isDateWithinBounds(requestDate, dateBounds)) {
+            urlContainer[name] = undefined;
+            return;
+        }
+
+        const defaultRequestConfig = {
             timeout: 10000,
-            followRedirect: true,
-            maxRedirects: 10
-        }, function (error, response, body) {
-            if (!error && response.statusCode == 200) {
-                if (typeof notReachable[name] != "undefined") {
-                  if (notReachable[name] == "notagain") {
-                    delete notReachable[name];
-                    var message = name + " " + emoji.get('beers') + " " + getFormattedDate();
-                    send(message);
-                    console.log(message);
-                  }
-                }
+            followRedirect: true
+        };
+
+        const requestConfigWithUrl = Object.assign(
+            { url: params.url },
+            params.requestConfig ?? defaultRequestConfig
+        )
+
+        request.get(requestConfigWithUrl, function (error, response, body) {
+            const responseDate = new Date();
+
+            if (error !== null) {
+                handleLocalError(name, responseDate);
+            } else if (response.statusCode == 200) {
+                handleSuccessStatus(name, params.url, responseDate);
             } else {
-                if (typeof notReachable[name] == "undefined") { // first time
-                    notReachable[name] = "true";
-                    var message = name + " " + emoji.get('hankey') + " " + getFormattedDate();
-                    console.log("first time :" + message);
-
-                } else if (notReachable[name] == "true") { // secondtime
-                    notReachable[name] = "notagain"; // we don't want second notification
-                    var message = name + " " + emoji.get('hankey') + " " + getFormattedDate();
-                    console.log(message);
-                    send(message);
-                }
+                handleDownStatus(name, params.url, responseDate, params.mentionsWhenDown);
```

Comment:
just pass `params, responseDate` and handle it in there, no need to pass 2 params.XXX